### PR TITLE
Type use fix

### PR DESCRIPTION
**Pull request by vietj**
This PR provides support for obtaining better `@Nullable` information in codegen types. The current support relies on the `javax.lang.model` which is not complete and fails to obtain the annotation with parameterized types (e.g `Handler<AsyncResult<@Nullable String>>`). This is not really a bug but rather an incomplete implementation of the API that is implemented in Java 9.

This PR relies on internal types of the compiler to extract the relevant information. As this is Java 8 specific, the implementation falls back on the `javax.lang.model` API when the internal API is not available and therefore should work with Java 9. When Java 9 will be released we will make sure it works as expected. Note this is not a runtime issue but rather a codegen compile time issue to address.

### CODE REVIEW

**Comment by tsegismont** [comment]
Is the first commit related to an existing GitHub issue?

**Comment by vietj** [comment]
no it's not

**Comment by vietj** [comment]
actually yes :-) https://github.com/vert-x3/vertx-codegen/issues/111

**Comment by tsegismont** [comment]
You mean #111 is fixed by this whole PR, right? I was only talking about the first commit (_Fix for vertx-service-proxies_)

File: pom.xml [hunk 1]
```diff
@@ -231,4 +221,29 @@
...
     </plugins>
   </build>
+
+  <profiles>
+    <profile>
+      <id>tools</id>
+      <properties>
+        <tools.jar>${java.home}/../lib/tools.jar</tools.jar>
+      </properties>
+      <activation>
```
Comment (tsegismont) [inline pom.xml]: Why use a property instead of jdk version activation?

**Comment by vietj** [comment]
yes, I've removed the streaming proxy experiments branch

**Comment by vietj** [comment]
I would have preferred to merge the branch myself as this one goes with RxJava2 review

**Comment by tsegismont** [comment]
Ha, sorry

2017-07-24 22:14 GMT+02:00 Julien Viet <notifications@github.com>:

**Pull request merged**